--- a/gdal.cmake
+++ b/gdal.cmake
@@ -619,8 +619,10 @@
 if (MSVC)
   target_sources(${GDAL_LIB_TARGET_NAME} PRIVATE gcore/Version.rc)
   source_group("Resource Files" FILES gcore/Version.rc)
-  if (CMAKE_CL_64)
+  if (CMAKE_SYSTEM_PROCESSOR STREQUAL AMD64)
     set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
+  elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL ARM64)
+    set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES STATIC_LIBRARY_FLAGS "/machine:arm64")
   endif ()
 endif ()
 
